tests/unit/models/transactions/test_better_transaction_flags.py

```diff
@@ -0,0 +1,413 @@
+            second=signed_expected,
+        )
+
+    def test_trust_set_flags(self):
+        amnt = models.IssuedCurrencyAmount(
+            currency="USD",
+            issuer="rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B",
+            value="1",
+        )
+        actual = models.TrustSet(
+            account=ACCOUNT,
+            limit_amount=amnt,
+            flags=models.TrustSetFlagInterface(
+                tf_clear_freeze=True,
+                tf_clear_no_ripple=True,
+                tf_set_auth=True,
+                tf_set_freeze=True,
+                tf_set_no_ripple=True,
+            ),
+        )
+        self.assertTrue(actual.has_flag(flag=0x00010000))
+        self.assertTrue(actual.is_valid())
+        flags = models.TrustSetFlag
+        expected = models.TrustSet(
+            account=ACCOUNT,
+            limit_amount=amnt,
+            flags=[*flags],
+        )
+        signed_actual = safe_sign_transaction(
+            transaction=actual,
+            wallet=WALLET,
+        )
+        signed_expected = safe_sign_transaction(
+            transaction=expected,
+            wallet=WALLET,
+        )
+        self.assertEqual(
+            first=signed_actual,
+            second=signed_expected,
+        )
+
+    def test_enable_amendment(self):
+        amdmt = "42426C4D4F1009EE67080A9B7965B44656D7714D104A72F9B4369F97ABF044EE"
+        seq = 21225473
+        actual = models.EnableAmendment(
+            amendment=amdmt,
+            ledger_sequence=seq,
+            flags=models.EnableAmendmentFlagInterface(
+                tf_got_majority=True,
+                tf_lost_majority=True,
+            ),
+        )
+        flags = models.EnableAmendmentFlag
+        expected = models.EnableAmendment(
+            amendment=amdmt,
+            ledger_sequence=seq,
+            flags=[*flags],
+        )
+        signed_actual = safe_sign_transaction(
+            transaction=actual,
+            wallet=WALLET,
+        )
+        signed_expected = safe_sign_transaction(
+            transaction=expected,
+            wallet=WALLET,
+        )
+        self.assertEqual(
+            first=signed_actual,
+            second=signed_expected,
+        )
+
+    def test_no_flags_defined(self):
+        try:
+            cancel_offer = models.OfferCancel(
+                account=ACCOUNT,
+                offer_sequence=6,
+            )
+            safe_sign_transaction(
+                transaction=cancel_offer,
+                wallet=WALLET,
+            )
+            dest = "ra5nK24KXen9AHvsdFTKHSANinZseWnPcX"
+            amnt = "10000000"
+            pymnt = models.Payment(
+                account=ACCOUNT,
+                destination=dest,
+                amount=amnt,
+            )
+            safe_sign_transaction(
+                transaction=pymnt,
+                wallet=WALLET,
+            )
+        except XRPLModelException:
+            self.fail("Could not sign transaction without flags")
```

Comment:
Oh, actually if no flags are defined, it should default to flags = 0 (It's an optional parameter)

Reply:
How about: "If no flags are defined 'flags' have to be 0. Txn could not be signed."?

Reply:
That works. Or maybe "Failed to sign a transaction which had no flags defined."